buildkite/scripts/test-nix.sh

```diff
@@ -47,19 +48,19 @@ nix-env -i git-lfs
 
 git config --global --add safe.directory /workdir
 
-git fetch origin $1:$1
-# Nix has issue when performing operations on detached head
+# We are in buildkite context so all buildkite related envs are available
+# We can use BUILDKITE_BRANCH to checkout the PR branch
+# Checking out the PR branch is necessary to make nix happy as it doesn't like detached head.
+# To be more precise nix has issue when performing operations on detached head
 # On Ci machine it spit out issues like:
 # fatal: reference is not a tree: ....
 # error:
 #       … while fetching the input 'git+file:///workdir'
 #
 #       error: program 'git' failed with exit code 128
 # That is why we checkout branch explicitly
-git checkout $1
 
-git submodule sync
```

Comment:
Agree that the previous solution was ugly.

I used the following in experimental nix pipeline:

```bash
git branch -D $BUILDKITE_BRANCH 2>/dev/null || true
git checkout -b $BUILDKITE_BRANCH
git reset --hard $BUILDKITE_COMMIT
```

Current solution looks like an overkill to me. Buildkite sets up the right commit without a trouble, the issue that nix complains about is the branch name. Solution above ensures the commit and branch are set as expected.